overlord/devicestate/devicestate.go

```diff
@@ -491,6 +492,11 @@ func Remodel(st *state.State, new *asserts.Model) (*state.Change, error) {
 		return nil, fmt.Errorf("cannot remodel to different gadgets yet")
 	}
 
+	// TODO: should we run a remodel only while no other change is
+	// running?  do we add a task upfron that waits for that to be
+	// true? Do we do this only for the more complicated cases
+	// (anything more than adding required-snaps really)?
```

Comment:
Without spending too much time thinking about bigger picture of this question, I'd say that - given the specific uses of this feature and the fact that is generally going to be used once (is it a sensible assumption?), I'd strive to keep it simple (e.g. remodel only when no other changes are in flight) and not try to optimize, at least for the initial version.